src/push.go

```diff
@@ -42,6 +43,7 @@ func (f *PushOnlyFlags) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.ActionsAdminUser, "actions-admin-user", "", "A user to impersonate for the push requests. To use the default name, pass 'actions-admin'. Note that the site_admin scope in the token is required for the impersonation to work.")
 	cmd.Flags().StringVar(&f.Token, "destination-token", "", "Token to access API on GHES instance")
 	cmd.Flags().BoolVar(&f.DisableGitAuth, "disable-push-git-auth", false, "Disables git authentication whilst pushing")
+	f.IsAE = false
```

Comment:
I think we should also add a command line flag option here for the case customer is running against GHAE without impersonation.

Reply:
why de we want this to be command user input, when we already able to find via
`flags.IsAE = rootResponse.Header.Get(enterpriseVersionHeaderKey) == enterpriseAegisVersionHeaderValue`

Reply:
🤔 The first call we do within `Push` without any conditions is the [Create Repo](https://github.com/actions/actions-sync/blob/6d732db1cf8e12f2bb06adf27a15bafd368caf89/src/push.go#L219) call. So if you opt out of the impersonation the first time we would be able to catch that information is from that call. Would you then suggest to check the header value in the case we [get the 422 result](https://github.com/actions/actions-sync/blob/6d732db1cf8e12f2bb06adf27a15bafd368caf89/src/push.go#L222) and then retry it with `internal` visibility?

Reply:
We discussed offline, here are the points:
- Remove the newly added flag
- In repo created code flow, currenlty we are creating anf if 422, then fetching
  - instead we will first fetch the repo 
  - and if 404, then we will check the response header if ghae or not
  - and create repo (public by default and internal if its ghae)

Reply:
I actually missed that we retrieve the user before retrieving/creating the repo 🤦 We can get the appliance information from there. Anyway changed the behavior and added E2E tests for the impersonation scenario and to check the repo visibility.

Reply:
Thats cool.